File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:

        n=len(cardPoints)
        s=sum(cardPoints)
        mini_sum=sum(cardPoints[:n-k])
        res=mini_sum
        l,r=0,n-k
        while r<n:
            res+=(cardPoints[r]-cardPoints[l])
            l+=1
            r+=1
            mini_sum=min(mini_sum,res)

        return s-mini_sum


        

File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.js
/**
 * @param {number[]} cardPoints
 * @param {number} k
 * @return {number}
 */
var maxScore = function(cardPoints, k) {

    const n =cardPoints.length;
    let l=0;
    let r=n-k;
    function sum(nums,j){
        let sumi=0;
        for(let i=0;i<j;i++)
        {
            sumi+=nums[i];

        }
        return sumi
    }
    let mini_sum=sum(cardPoints,n-k);
    let res=mini_sum
    while(r<n)
    {
        res+=(cardPoints[r]-cardPoints[l]);
        l++;
        r++;
        mini_sum=Math.min(mini_sum,res);
        
    }

    return sum(cardPoints,n)-mini_sum;


    
};